File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/pr/PRWaveTrigger.java
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.pr;

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveMetadata;
import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveTrigger;

/**
 * The trigger that engine the heartbeat wave P.
 */
public class PRWaveTrigger extends WaveTrigger<PRWaveMetadata, WaveMetadata> {
    @Override
    public PRWaveMetadata primeMetadata() {
        return new PRWaveMetadata();
    }
}

File: src/main/java/com/github/cao/awa/majalis/env/human/organ/heart/heartbeat/wave/pr/PRWaveTrigger.kt
package com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.pr

import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveMetadata
import com.github.cao.awa.majalis.env.human.organ.heart.heartbeat.wave.WaveTrigger

/**
 * The trigger that engine the heartbeat wave PR.
 */
class PRWaveTrigger : WaveTrigger<PRWaveMetadata, WaveMetadata>() {
    override fun primeMetadata(): PRWaveMetadata = PRWaveMetadata()
}
